src/coord/src/coord.rs

```diff
@@ -954,6 +955,11 @@ impl<S: Append + 'static> Coordinator<S> {
                 );
             }
             Err(e) => {
+                if matches!(e, CoordError::ReadThenWriteTimeoutExceeded(..)) {
+                    // Clean up any state that might get left behind from e.g. a
+                    // forced cancellation due to a timeout.
+                    self.force_cancel_cleanup(session.conn_id()).await;
```

Comment:
Minor, but this technically cancels too much. It probably wants to cancel only the one peek involved in the read-then-write op that failed, rather than other portals that are live.